src/IPC/Queue.h

```diff
@@ -0,0 +1,9 @@
+#define ProjectId 123
+#define PathName "send_main"
+
+typedef struct {
+    long index;                 
+    long endIndex;
+    int sizeMessage;
+    char payload[BUFFERSIZE];   /* bytes in the message */
```

Comment:
You could use char vector instead of char array, it is more C++ style, you can do the same things than with char array and even more.